### PR TITLE
[WebServerBundle] Added ability to display the current hostname address if available when binding to 0.0.0.0

### PR DESCRIPTION
**Pull request by respinoza**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | ? (no tests for server commands)
| Fixed tickets | #28585
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Tries to get the local hostname into an ip (if available) using the PHP gethostname* methods.

### CODE REVIEW

File: src/Symfony/Bundle/WebServerBundle/WebServerConfig.php [hunk 1]
```diff
@@ -101,6 +101,22 @@ public function getAddress()
         return $this->hostname.':'.$this->port;
     }
 
+    /**
+     * @return string contains resolved hostname if available, empty string otherwise
+     */
+    public function getExtraAddress()
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/WebServerBundle/WebServerConfig.php]: What about renaming this method as `getRealAddress()` or `getResolvedAddress()` ?

File: src/Symfony/Bundle/WebServerBundle/Command/ServerRunCommand.php [hunk 2]
```diff
@@ -132,7 +132,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $server = new WebServer();
             $config = new WebServerConfig($documentRoot, $env, $input->getArgument('addressport'), $input->getOption('router'));
 
-            $io->success(sprintf('Server listening on http://%s', $config->getAddress()));
+            $message = sprintf('Server listening on http://%s', $config->getAddress());
+            if ('' !== $extra = $config->getExtraAddress()) {
+                $message = sprintf('%s (http://%s)', $message, $extra);
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/WebServerBundle/Command/ServerRunCommand.php]: Not sure if we can hardcode `http://` here. What if we are using `https://` in the local environment? Maybe it's better to detect that and add it in the `getExtraAddress()` method.

Comment (javiereguiluz) [inline src/Symfony/Bundle/WebServerBundle/Command/ServerRunCommand.php]: To make it a bit easier to understand for users, we could change the output from this:

```
[OK] Server listening on http://0.0.0.0:8080 (http://172.20.0.2:8080)
```

to something like this:

```
[OK] Server listening on http://0.0.0.0:8080 (resolved as http://172.20.0.2:8080)
```

Comment (respinoza) [reply src/Symfony/Bundle/WebServerBundle/Command/ServerRunCommand.php]: @javiereguiluz will do the change as `resolved`, this should also match the method name change (more consistent)

File: src/Symfony/Bundle/WebServerBundle/WebServerConfig.php [hunk 3]
```diff
@@ -101,6 +101,22 @@ public function getAddress()
         return $this->hostname.':'.$this->port;
     }
 
+    /**
+     * @return string contains resolved hostname if available, empty string otherwise
+     */
+    public function getResolvedAddress()
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebServerBundle/WebServerConfig.php]: I'd suggest calling it getDisplayAddress actually (nothing is "resolved" to me)

Comment (javiereguiluz) [reply src/Symfony/Bundle/WebServerBundle/WebServerConfig.php]: Do you recommend to reword the message displayed to users too? Currently it says `... (resolved as ...)`  I think it's easy to understand that way ... but may there's a better wording. Thanks!

**Comment by respinoza** [comment]
Updated with the new messages, thanks to @jkufner suggestion, I believe the message now looks cleaner and conveys what we want.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @respinoza.

**Pull request merged**